gnd/src/main/java/com/google/android/gnd/system/StorageManager.java

```diff
@@ -120,35 +116,24 @@ private File getLocalFileFromDestinationPath(String destinationPath)
     return fileUtil.getFile(splits[splits.length - 1]);
   }
 
+  private Uri getFileUriFromDestinationPath(String destinationPath) throws FileNotFoundException {
+    File file = getLocalFileFromDestinationPath(destinationPath);
+    return Uri.fromFile(file);
+  }
+
   /**
    * Load photo from the provided destination path.
    *
    * <p>If the image is not uploaded yet, then parse the filename from path and load the file from
    * local storage.
    *
-   * @param imageView Placeholder for photo field
    * @param destinationPath Destination path of the uploaded photo
    */
-  public void loadPhotoFromDestinationPath(ImageView imageView, String destinationPath) {
+  public void loadPhotoFromDestinationPath(String destinationPath, Callback callback) {
     firestoreStorageManager
         .getDownloadUrl(destinationPath)
-        .doOnSuccess(
-            uri -> {
-              // Load the file from Firestore Storage
-              Picasso.get()
-                  .load(uri)
-                  .placeholder(R.drawable.ic_photo_grey_600_24dp)
-                  .into(imageView);
-            })
-        .doOnError(
-            throwable -> {
-              // Load file locally
-              File file = getLocalFileFromDestinationPath(destinationPath);
-              Picasso.get()
-                  .load(file)
-                  .placeholder(R.drawable.ic_photo_grey_600_24dp)
-                  .into(imageView);
-            })
+        .doOnSuccess(callback::run)
+        .doOnError(throwable -> callback.run(getFileUriFromDestinationPath(destinationPath)))
         .subscribe();
```

Comment:
Sorry if I mentioned it before, but this probably won't work. Since you need to properly manage the returned disposable, we only subscribe to streams in lifecycle-aware contexts like Fragments. The right thing to do here I believe, would be to return a Single<Uri>, which can be created using RxTask.

Reply:
Thanks for catching this! This simplified the code greatly.